src/java.management/share/classes/com/sun/jmx/remote/internal/ServerNotifForwarder.java

```diff
@@ -343,10 +344,13 @@ public void terminate() {
     //----------------
     // PRIVATE METHODS
     //----------------
-
     @SuppressWarnings("removal")
     private Subject getSubject() {
-        return Subject.getSubject(AccessController.getContext());
+        if (!SharedSecrets.getJavaLangAccess().allowSecurityManager()) {
+            return Subject.current();
+        } else {
+            return Subject.getSubject(AccessController.getContext());
+        }
```

Comment:
`Subject.current()` should work for both cases. See the impl of it.

Reply:
ok!

Reply:
OK if it was actually getting specifically the ACC to use it maybe, but this just needs to resolve a Subject.